server/resource_server/utils/verification.go

```diff
@@ -0,0 +1,96 @@
+package utils
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+	"github.com/consensys/gnark/frontend"
+	"log"
+	"unicode/utf8"
+)
+
+const VerificationCodeSize = 6
+const EmailFrRepresentationSize = 38
+
+const runesPerElement = 7
+const bytes = 4
+const elementByteSize = 32
+
+func StringToFrElements(input string) [EmailFrRepresentationSize]fr.Element {
```

Comment:
This function is also a piece of work. I need to send you a screen shot of my notes during the plane ride review... I'll do it on Discord